backend/route/classdata_test.go

```diff
@@ -0,0 +1,190 @@
+package route
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"sort"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+
+	"github.com/ucsdscheduleplanner/UCSD-Schedule-Planner/backend/store"
+)
+
+var getClassDataColumns = []string{
+	"Department",
+	"CourseNum",
+	"SectionId",
+	"CourseId",
+	"ClassType",
+	"Days",
+	"Time",
+	"Location",
+	"Room",
+	"Instructor",
+	"Description"}
+
+func TestBuildQueryClassData(t *testing.T) {
+	testRequests := [][]ClassDataRequest{
+		{{"SP19", "LTEN", "25"}},
+		{{"SP19", "LTEN", "26"}, {"SP19", "LTEN", "26"}},
+		{{"SP19", "LTEN", "26"}, {"FA20", "LTEN", "26"}},
+	}
+
+	// note: there is no order of quarters and queryParams guaranteed
+	// use same class to make testing easier
+
+	expects := []struct {
+		queryRex    string
+		quarters    []string
+		queryParams []interface{}
+	}{
+		{
+			fmt.Sprintf(
+				`SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\)\)`, classDataColumns, "SP19",
+			),
+			[]string{"SP19"},
+			[]interface{}{"LTEN", "25"},
+		},
+		{
+			fmt.Sprintf(
+				`SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\),\(\?,\?\)\)`, classDataColumns, "SP19",
+			),
+			[]string{"SP19"},
+			[]interface{}{"LTEN", "26", "LTEN", "26"},
+		},
+		{
+			fmt.Sprintf(
+				`SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\)\) UNION ALL \(SELECT %s FROM %s WHERE \(DEPARTMENT, COURSE_NUM\) IN \(\(\?,\?\)\)\)`, classDataColumns, "(SP19|FA20)", classDataColumns, "(SP19|FA20)",
+			),
+			[]string{"SP19", "FA20"},
+			[]interface{}{"LTEN", "26", "LTEN", "26"},
+		},
+	}
+
+	for i, testRequest := range testRequests {
+
+		expected := expects[i]
+
+		rQuery := regexp.MustCompile(expected.queryRex)
+
+		gotQuery, gotQuarters, gotParams := buildQueryClassData(testRequest)
+
+		if !rQuery.MatchString(gotQuery) {
+			t.Errorf("Wrong sql query, want '%v' got '%v'", rQuery, gotQuery)
+		}
+
+		if len(gotQuarters) != len(expected.quarters) {
+			t.Fatalf("Wrong size of quarters slice, want '%v' got '%v'", expected.quarters, gotQuarters)
+		}
+
+		// order matters for testing but not matters for program
+		sort.Strings(gotQuarters)
+		sort.Strings(expected.quarters)
+
+		for j, s := range gotQuarters {
+			if s != expected.quarters[j] {
+				t.Errorf("Wrong quarter: Want '%v', got '%v' with mismatch at index %v", expected.quarters, gotQuarters, j)
+			}
+		}
+
+		if len(gotParams) != len(expected.queryParams) {
+			t.Fatalf("Wrong size of query params slice, want '%v' got '%v'", expected.queryParams, gotParams)
+		}
+
+		for j, s := range gotParams {
+			if s != expected.queryParams[j] {
+				t.Errorf("Wrong query params: Want '%v', got '%v' with mismatch at index %v", expected.queryParams, gotParams, j)
+			}
+		}
+
+	}
+
+}
+
+func TestGetClassData(t *testing.T) {
```

Comment:
More tests for get class data and departments and such, these are the big ones that the web app uses.

Reply:
Will add after a consensus of REST API